starboard/BUILD.gn

```diff
@@ -98,16 +100,21 @@ group("starboard") {
       "//third_party/llvm-project/libunwind:unwind_evergreen",
       "//third_party/musl:c",
     ]
+    if (build_with_separate_cobalt_toolchain) {
+      data_deps = [ ":starboard_platform_group($starboard_toolchain)" ]
```

Comment:
this comes under the sb_is_evergreen conditional 
`build_with_separate_cobalt_toolchain` should be false under that or shouldnt matter..? so the if condition feels irrelevant in this case.

if (sb_is_evergreen) {
    public_deps += [
      "//starboard/elf_loader:sabi_string",
      "//third_party/llvm-project/compiler-rt:compiler_rt",
      "//third_party/llvm-project/libcxx:cxx",
      "//third_party/llvm-project/libcxxabi:cxxabi",
      "//third_party/llvm-project/libunwind:unwind_evergreen",
      "//third_party/musl:c",
    ]
    if (build_with_separate_cobalt_toolchain) {
      data_deps = [ ":starboard_platform_group($starboard_toolchain)" ]
    }
  } else {

Reply:
`build_with_separate_cobalt_toolchain` isn't directly related to `sb_is_evergreen`. We can have evergreen builds without `build_with_separate_cobalt_toolchain=true` right now, though we'll want to clean that up eventually to they mean the same thing